bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java

```diff
@@ -260,6 +260,12 @@ public void enableChannel(ChannelUID channelUID) {
                     }
                     item.addTags(channelType.getTags());
                     item.setCategory(channelType.getCategory());
+                    if (channelType.getState().getPattern() != null) {
+                        String label = channelType.getLabel() + " [" + channelType.getState().getPattern() + "]";
```

Comment:
> It should be in the responsibility of the UI to render at the end a label, which combines real label and meta data like the pattern

This is currently not the case - as you can see in the demo.items, the formatting is still part of the label and it is required by e.g. the Classic UI. Btw, this PR is all about fixing the code to make it work with the Classic UI, which is completely broken at the moment as it has not been considered in the past refactorings. And I expect the same for all other REST-based clients besides Paper UI.
I would therefore suggest to do it at the moment this way, until we might have implemented https://www.eclipse.org/forums/index.php/t/1027209/.